core/links.py

```diff
@@ -0,0 +1,190 @@
+    for link in new_sv_links:
+        if not link in before_sv_links:
+            if not link.from_socket_id in before_output_sockets:
+                if not link.from_node_id in affected_nodes:
+                    affected_nodes.append(link.from_node_id)
+            if not link.to_node_id in affected_nodes:
+                affected_nodes.append(link.to_node_id)
+    return affected_nodes
+
+
+def get_new_unlinked_nodes(before_inputted_nodes, before_input_sockets, input_sockets, nodes_dict):
+    affected_nodes = []
+
+    for node_id, socket in zip(before_inputted_nodes, before_input_sockets):
+        if not socket in input_sockets:
+            #if the node has been deleted it is not affected
+            if node_id in nodes_dict:
+                affected_nodes.append(node_id)
+
+    return affected_nodes
+
+
+def split_new_links_data(new_sv_links):
+    inputted_nodes = []
+    input_sockets = []
+    output_sockets = []
+    for link in new_sv_links:
+        inputted_nodes.append(link.to_node_id)
+        input_sockets.append(link.to_socket_id)
+        output_sockets.append(link.from_socket_id)
+
+    return  inputted_nodes, input_sockets, output_sockets
+
+
+def bl_links_to_sv_links(node_tree):
+
+    new_sv_links = []
+    for link in node_tree.links:
+        if not link.to_socket.node.bl_idname == 'NodeReroute':
+            #recursive function to override reroutes
+            output_socket, output_node = get_output_socket_id(link.from_socket)
+            if output_socket:
+                sv_link = SvLink(
+                    from_node_id=output_node,
+                    to_node_id=link.to_socket.node.node_id,
+                    from_socket_id=output_socket,
+                    to_socket_id=link.to_socket.socket_id )
+                new_sv_links.append(sv_link)
+
+    return new_sv_links
+
+
+def fill_links_memory(node_tree):
+    tree_id = node_tree.tree_id
+    new_sv_links = bl_links_to_sv_links(node_tree)
+
+    inputted_nodes, input_sockets, output_sockets = split_new_links_data(new_sv_links)
+
+    sv_links_cache[tree_id] = new_sv_links
+    sv_linked_output_sockets_cache[tree_id] = output_sockets
+    sv_linked_input_sockets_cache[tree_id] = input_sockets
+    sv_linked_inputted_nodes_cache[tree_id] = inputted_nodes
+
+
+def get_affected_groups(node_tree):
+    affected_groups = []
+    for node in node_tree.nodes:
+        if 'SvGroupNode' in node.bl_idname:
+            subtree = node.monad
+            tree_id = subtree.tree_id
+            fill_memory_is_ready = tree_id in sv_links_cache
+            if fill_memory_is_ready:
+                has_been_updated = use_link_memory(subtree)
+            else:
+                fill_links_memory(subtree)
+                subtree.has_changed = True
+                node.process()
+                has_been_updated = True
+
+            if has_been_updated:
+                affected_groups.append(node)
+    return affected_groups
+
+
+def use_link_memory(node_tree):
```

Comment:
Does not do updates from this method. We now have `SverchCustomTree` class responsible for updates. We should centralize all updates in one place so it would be easier to deal with update events.
Don't return any `bool` values if purpose of the method is not in doing some test.
Most likely you should create only affected nodes in this method all other code should be moved somewhere.